include/myst/spinlock.h

```diff
@@ -78,6 +79,8 @@ MYST_INLINE void myst_spin_lock(myst_spinlock_t* spinlock)
 
 MYST_INLINE void myst_spin_unlock(myst_spinlock_t* spinlock)
 {
+    // Assert that the spinlock is locked.
+    myst_assume(*spinlock != 0);
```

Comment:
Instead of storing `1` in the lock, we could store the 4 bytes of the kernel thread pointer (GS[0]). That would allow us to assert that only the thread that owns the spinlock actually unlocks it.' 

Cases that would go undetected would be if two threads have the same 4-byte identifier. That is still better than not detecting any ownership issues.